contracts/genie-airdrop/src/contract.rs

```diff
@@ -28,17 +28,27 @@ pub fn instantiate(
             "to_timestamp must be greater than from_timestamp",
         ));
     }
+    if msg.allocated_amounts.iter().any(|&x| x == Uint128::zero()) {
+        return Err(StdError::generic_err(
+            "allocated_amounts must not contain zero",
+        ));
+    }
+    if msg.allocated_amounts.is_empty() {
+        return Err(StdError::generic_err("allocated_amounts must not be empty"));
+    }
```

Comment:
not asking to change, but interesting, normally i would check for empty first before loop through the elements 🤣